tools/shared.py

```diff
@@ -2356,6 +2356,11 @@ def get_subresource_location(path, data_uri=None):
     else:
       return os.path.basename(path)
 
+  # Strips all preprocessor directives from a string of JavaScript
+  @staticmethod
+  def strip_preprocessor_directives(s):
+    return re.sub('#if .*?\n.*?\n#endif', '', s, flags=re.DOTALL)
```

Comment:
Hm, well, would you want the whole preprocessor reimplemented in Python, or just a temporary partial impressions? Maybe I could do a Python regex version that just handles the specific cases needed for the HTML to fully work and a separate issue could be created for swapping out that workaround with a hook into the real preprocessor.

As far as the conditions needed, not 100% sure offhand, but I think it's just ASSERTIONS.